Sources/MapboxCoreNavigation/NavigationEventsManager.swift

```diff
@@ -129,6 +135,7 @@ open class NavigationEventsManager {
         var event = ActiveNavigationEventDetails(dataSource: dataSource, session: sessionState, defaultInterface: usesDefaultUserInterface)
         event.event = MMEEventTypeNavigationCancel
         event.arrivalTimestamp = sessionState.arrivalTimestamp
+        event.appMetadata = userInfo
```

Comment:
You can use the initializer above to provide userInfo (and for similar instantiations of `ActiveNavigationEventDetails` and `PassiveNavigationEventDetails`)